### PR TITLE
feat: Add Other Files for iOS

### LINKED ISSUE #1199: iOS | Add --other-files
**Author the user story for this feature**

As an iOS developer, I want to push files to the device so I can run tests against them.

**Describe the solution**

This is an unpublished feature at the moment.

```java
/**
   * List of files to push to the device before starting the test.
   * @param pushFiles pushFiles or {@code null} for none
   */
  public IosTestSetup setPushFiles(java.util.List<IosDeviceFile> pushFiles) {
    this.pushFiles = pushFiles;
    return this;
  }
```

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #1199

Other files could be sent using `--other-files` CLI or `other-files` YAML config for iOS

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

**Review by adamfilipow92 — APPROVED** [review]

**Comment by pawelpasterz** [comment]
It would be nice to have the possibility of testing it on the actual app. If it's doable, please update the test plan

File: docs/index.md [hunk 1]
```diff
@@ -146,6 +146,15 @@ gcloud:
   #   locale: es_ES
   #   orientation: landscape
 
+
+  ## A list of device-path=file-path pairs that specify the paths of the test device and the files you want pushed to the device prior to testing.
```
[the remark this answers is not in the document]
Comment (piotradamczyk5) [reply docs/index.md:150]: This is copy paste from google page

**Comment by piotradamczyk5** [comment]
@Mergifyio rebase

**Review by Sloox — APPROVED** [review]

File: docs/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sloox) [reply docs/index.md:150]: I think leave it if its copy pasted. But yes you are correct.

**Comment by Sloox** [comment]
> It would be nice to have the possibility of testing it on the actual app. If it's doable, please update the test plan

integration tests? :)

**Comment by pawelpasterz** [comment]
> > It would be nice to have the possibility of testing it on the actual app. If it's doable, please update the test plan
> 
> integration tests? :)

I'll try to cover it within https://github.com/Flank/flank/issues/1212

**Comment by piotradamczyk5** [comment]
@Mergifyio refresh

**Pull request merged**